lpcompiler.pas

```diff
@@ -2579,20 +2618,25 @@ function TLapeCompiler.ParseExpression(ReturnOn: EParserTokenSet = []; FirstNext
         else
           Op := op_Unknown;
 
-        if (not (Op in AssignOperators)) and MethodType(Node.resType()) then
+        if (not (Op in AssignOperators)) and __IsProperty(Node.resType()) then
+        begin
+          Result := TLapeTree_InvokeProperty.Create(Node, Node);
+          TLapeTree_InvokeProperty(Result).PropertyType := ptRead;
+        end
+        else if (not (Op in AssignOperators)) and MethodType(Node.resType()) then
           Result := TLapeTree_Invoke.Create(Node, Node)
-        else if (Op = op_Assign) and (lcoAutoProperties in Node.CompilerOptions) and MethodType(TLapeTree_Operator(Node).Left.resType()) then
+        else if (Op in AssignOperators) and __IsProperty(TLapeTree_Operator(Node).Left.resType()) then
         begin
-          Result := TLapeTree_Invoke.Create(TLapeTree_Operator(Node).Left, Node);
-          TLapeTree_Invoke(Result).addParam(TLapeTree_Operator(Node).Right);
+          Result := TLapeTree_InvokeProperty.Create(TLapeTree_Operator(Node).Left, Node);
+          TLapeTree_InvokeProperty(Result).addParam(TLapeTree_Operator(Node).Right);
+          TLapeTree_InvokeProperty(Result).PropertyType := ptWrite;
```

Comment:
No it can not. Because in the tree we check the result types, and see if they go along with the expected property type.
A for this is done in the tree:
`> TFoo.Foo[1, 2]` is ptRead, so the property called MUST have a result.
`> TFoo.Foo(index:Int32; value: Int32)` is ptWrite, it can't have a result.